scripts/tests/test_zkai_native_block_boundary_pivot_selector_gate.py

```
@@ -0,0 +1,132 @@
+            tsv_path = gate.pathlib.Path(tsv_handle.name)
+        try:
+            gate.write_outputs(payload, json_path, tsv_path)
+            self.assertEqual(json.loads(json_path.read_text(encoding="utf-8")), payload)
+            self.assertTrue(tsv_path.read_text(encoding="utf-8").startswith("\t".join(gate.ROW_COLUMNS)))
+        finally:
+            json_path.unlink(missing_ok=True)
+            tsv_path.unlink(missing_ok=True)
+
+    def test_write_outputs_rejects_outside_evidence_dir(self) -> None:
+        with tempfile.TemporaryDirectory() as tmpdir:
+            with self.assertRaisesRegex(gate.PivotSelectorError, "evidence dir"):
```

Ruff (0.15.12)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)